staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation_test.go

```diff
@@ -339,6 +339,17 @@ func TestValidationExpressions(t *testing.T) {
 				"self.val1 + [4, 5] == [1, 2, 3, 4, 5]",
 			},
 		},
+		{name: "string lists",
```

Comment:
do we need a negative test that `obj: objs([]interface{}{"a", 1, "c"}),` errors when join is attempted?

Reply:
The CEL compiler prevents mixed type lists via https://github.com/google/cel-go/blob/d39523c445fcf548ef6529b63c1a1045cc373d37/cel/options.go#L129, which we have enabled.

I could test that non-string lists are not eligible to be joined, but it seems excessive to test that all functions are only available on the receivers that they are defined for.

Reply:
ok, if we guard against it and test for that guard, that's fine.

Reply:
SG. I've added the tests for the guard (for both lists and maps).